conu/utils/__init__.py

```diff
@@ -109,3 +114,112 @@ def mkdtemp():
+            if not found:
+                logger.error("`%s` exited with a non-zero return code (%s)",
+                             noop_invocation, p.returncode)
+                logger.error("command stdout = %s", stdout)
+                logger.error("command stderr = %s", stderr)
+    if not found:
+        raise CommandDoesNotExistException(exc_msg)
+    return True
+
+
+def s2i_command_exists():
+    return command_exists(
+        "s2i",
+        ["s2i", "version"],
+        "s2i command doesn't seem to be available on your system. Usually it's available in "
+        "'source-to-image' package. For more info, please consult the upstream documentation "
+        "available at 'https://github.com/openshift/source-to-image'."
+    )
+
+
+def atomic_command_exists():
+    return command_exists(
+        "atomic",
+        ["atomic", "-v"],
+        "atomic command doesn't seem to be available on your system. Usually it's available in "
+        "'atomic' package. For more info, please consult the upstream documentation "
+        "available at 'https://github.com/projectatomic/atomic'."
+    )
+
+
+def chcon_command_exists():
+    return command_exists(
+        "chcon",
+        ["chcon", "--version"],
+        "chcon command doesn't seem to be available on your system. Usually it's available in "
+        "'coreutils' package. Please consult documentation of your operating system."
+    )
+
+
+def setfacl_command_exists():
+    return command_exists(
+        "setfacl",
+        ["setfacl", "-v"],
+        "setfacl command doesn't seem to be available on your system. Usually it's available in "
+        "'acl' package. Please consult documentation of your operating system."
+    )
+
+
+def getenforce_command_exists():
+    return command_exists(
+        "getenforce",
+        ["getenforce"],
+        "getenforce command doesn't seem to be available on your system or SELinux is "
+        "misconfigured. Please consult documentation of your operating system."
+    )
+
+
+def check_docker_command_works():
+    """
+    Verify that dockerd and docker binary works fine. This is performed by calling `docker
+    version`, which also checks server API version.
+
+    :return: bool, True if all is good, otherwise ConuException or CommandDoesNotExistException
+              is thrown
+    """
+    try:
+        out = subprocess.check_output(["docker", "version"], stderr=subprocess.STDOUT)
+    except OSError:
+        logger.info("docker binary is not available")
+        raise CommandDoesNotExistException(
+            "docker command doesn't seem to be available on your system. "
+            "Please install and configure docker."
+        )
+    except subprocess.CalledProcessError as ex:
+        logger.error("exception: %s", ex)
+        logger.error("rc: %s, output: %r", ex.returncode, ex.output)
+        raise ConuException(
+            "`docker version` call failed, it seems that your docker daemon is misconfigured or "
```

Comment:
I really like the error description and it would be great to add something similar to api failures when dockerd is not running.